Issue#74 Network Availability Helper

fknives opened the pull request:
Added Network Availability Helper as requested in #74 

Added An example usage into the App module.

**Tested the following scenarios:**
- Turned off WiFi and SIM Internet => showed properly Inactive
- Turned on WiFi with proper connection => showed properly Active
- Turned on SIM Internet with proper connection => showed properly Active
- Turned on WiFi, but set invalid Manual Proxy => showed properly InActive

Created separate module, because it needs `android.permission.ACCESS_NETWORK_STATE` permission.

fknives committed:
issue#74 create extension function with network capability

fknives committed:
issue#74 Add Network to Android-all and bump Version

fknives committed:
issue#74 Delete no longer used NetworkState

fknives committed:
issue#74 Add option to use ConnectivityManager directly

fknives committed:
issue#74 Add Example usage in app module

zsoltboldizsar commented on network/src/main/java/com/halcyonmobile/android/common/extensions/network/ContextExtension.kt:
```diff
@@ -0,0 +1,45 @@
+    }
+
+@get:RequiresApi(Build.VERSION_CODES.M)
+val ConnectivityManager.isNetworkActive: Boolean
```
It seems to me that we could hide this, what do you think?

fknives replied:
My thinking here was that, if your min API is above 23, you wouldn't use the compat + deprecated things anymore, just copy this instead

zsoltboldizsar commented on network/src/main/java/com/halcyonmobile/android/common/extensions/network/ContextExtension.kt:
```diff
@@ -0,0 +1,45 @@
+/**
+ * Checks if the device's active Network has internet capability.
+ *
+ * @return true if Device has Active Internet Available Network, false otherwise.
```
Minor, thing but I would probably rephrase this to: `@return true if device has Available Network and Active Internet, false otherwise.` as it implies both the order and the requirements we're checking. How does that sound to you?

zsoltboldizsar reviewed (changes requested):
Great to have these extremely commonly required extensions present in our library for both use and further reference! Thank you :thumbsup:!

fknives committed:
PR#76 Fix subset of PR comments

- typo in strings.xml
- better description for Documentation

fknives committed:
PR#76 Adjust public API of the library

Have a compat and nonCompat extension on the Context, so it is easier to just get the bits that one actually needs, while still being able to use with lower minAPI level as well if needed.
